File: matlab/baseline.m
function f = baseline(x,a)

x(isnan(x))=0;

if nargin == 1
temp = mean(x);
f = x-temp(ones(1,size(x,1)),:,:);
end


if nargin == 2
%     for j = 1:size(x,2)
%         for k = 1:size(x,3)
%             f(:,j,k) = x(:,j,k) - mean(x(~isnan(x(:,j,k)),j,k));
%         end
%     end
    temp = mean(x(a,:,:));
    f = x-temp(ones(1,size(x,1)),:,:);
end

File: ride/baseline.py
import numpy as np


def baseline(x, a=None):

    x[np.isnan(x)] = 0.0

    if a is None:
        temp = np.mean(x, axis=0)
        f = x - temp[np.newaxis, :]
    else:
        temp = np.mean(x[a, :], axis=0)
        f = x - temp[np.newaxis, :]

    return f
